xprof/btx_timeline.cpp

```diff
@@ -135,6 +148,34 @@ static void add_event_power(timeline_dispatch_t *dispatch, std::string hostname,
   track_event->set_counter_value(power);
 }
 
+static void add_event_computeEU(timeline_dispatch_t *dispatch, std::string hostname,
+                                uint64_t process_id, uint64_t thread_id, uintptr_t did,
+                                uint32_t subDevice, uint64_t timestamp, float activeTime) {
+  perfetto_uuid_t track_uuid = get_computeEU_track_uuuid(dispatch, hostname, process_id, did, subDevice);
+  auto *packet = dispatch->trace.add_packet();
+  packet->set_trusted_packet_sequence_id(10000);
+  packet->set_timestamp(timestamp);
+  auto *track_event = packet->mutable_track_event();
+  track_event->set_type(perfetto_pruned::TrackEvent::TYPE_COUNTER);
+  track_event->set_track_uuid(track_uuid);
+  track_event->set_name("computeEngine Usage");
+  track_event->set_double_counter_value(activeTime);
+}
+
+static void add_event_copyEU(timeline_dispatch_t *dispatch, std::string hostname,
+                             uint64_t process_id, uint64_t thread_id, uintptr_t did,
+                             uint32_t subDevice, uint64_t timestamp, float activeTime) {
+  perfetto_uuid_t track_uuid = get_copyEU_track_uuuid(dispatch, hostname, process_id, did, subDevice);
+  auto *packet = dispatch->trace.add_packet();
+  packet->set_trusted_packet_sequence_id(10000);
+  packet->set_timestamp(timestamp);
+  auto *track_event = packet->mutable_track_event();
+  track_event->set_type(perfetto_pruned::TrackEvent::TYPE_COUNTER);
+  track_event->set_track_uuid(track_uuid);
+  track_event->set_name("copyEngine Usage");
+  track_event->set_double_counter_value(activeTime);
+}
+
```

Comment:
All those function look the same, maybe we can refactor to avoid code duplication? 
Template + to pass the function `get_copyEU_track_uuuid | get_computeEU_track_uuuid`, or lambda

Reply:
Modified accordingly.